### ISSUE TITLE
Support more channels in `colortogray`

### ISSUE THREAD

**Issue by votti** [CellProfiler/CellProfiler#3598]
Our images can be stacks of more than 20 channels, but the current channel selection settings in the`colortogray` module only supports up to 20 channels (hardcoded: https://github.com/CellProfiler/CellProfiler/blob/613247de8b9227b17f9b2eb31b43ad86682aee7d/cellprofiler/modules/colortogray.py#L210).

Thus I would suggest two options:
A) Either increase the hard coded max channel number to e.g. 100 channels
B) Change the widget from `cps.Choice` to `cps.Integer`

I am happy to write a pull request for either option.

**Comment by 0x00b1** [comment]
Hi, @votti! Yeah. Good call. I don't think we imagined a world with more than 20 channels. 😆 

Please send a PR and I'll merge it.

**Comment by AetherUnbound** [comment]
I'm working on making it so that any arbitrary number of channels can be added (so we don't run into this issue in the future and so the dropdown for which channel isn't a mile long if it doesn't have to be!). I should have a PR out for that this weekend

**Comment by votti** [comment]
Thanks a lot for taking care of this!
Cheers!

**Comment by votti** [comment]
Hey there,

Having  been locked away for a month in the obligatory military service, I unfortunately never came around to test the new `ColorToGray` in the GUI, but just read the code which looked good and like an elegant solution.

Giving it a try now, I noticed a severe usability problem: If I want to extract e.g. channel plane 20, I need now to add 20 channels to be able to select it, which is far from optimal.

I now got around to change the module to using the `cps.Integer` widget, which seems to work. I will do a pull request soon.
